google_analytics/notebooks/load.ipynb

```diff
@@ -34,24 +34,189 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 64,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded file ../data/train.csv\n",
+      "Shape is: (903653, 55)\n"
+     ]
+    }
+   ],
+   "source": [
+    "train = load(\"../data/train.csv\")\n",
+    "\n",
+    "const_cols = [c for c in train.columns if train[c].nunique(dropna=False) == 1]\n",
+    "train = train.drop(const_cols, axis=1)\n",
+    "\n",
+    "# Cast target\n",
+    "train[\"transactionRevenue\"] = train[\"transactionRevenue\"].fillna(0).astype(float)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hypothesis: certain profiles should be naively assigned a zero predicted revenue\n",
+    "\n",
+    "Since the new problem description includes much less information at prediction time (no session info) it might be beneficial to only focus on visitors with a high probability to spend - and assign a zero prediction to everyone else.\n",
+    "\n",
+    "**Which profile is improbable enough to make a purchase, that it makes sense to make that prediction naively?**\n",
+    "\n",
+    "We think that visitors with a single visit that did not lead to a purchase during the training period (prior to December of the testing year) have a very high probability of not buying anything during December and January either. Let's test this assumption."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 65,
    "metadata": {},
    "outputs": [],
    "source": [
-    "train = load(\"../data/train.csv\", nrows=100000)\n",
-    "test = load(\"../data/test.csv\", nrows=10000)\n",
+    "x_train = train[(train[\"date\"] < '2016-12-01')]\n",
+    "y_train = train[(train[\"date\"] >= '2016-12-01') & (train[\"date\"] >= '2017-02-01')]"
```

Comment:
This relates to the comment above. If this comment is true, then y_train should be december 2017 and january 2018 (>=2017-12-01 and < '2018-02-01').

Reply:
However, I just realize that when we want to check this for train_x (August 2016 till Oktober 15, 2017) and train_y (December 2017 and January 2018), we cannot do this yet. This is because, at this moment, we only have train data from August 2016 until August 2017. Is this the reason why you checked it for December 2016 and January 2017? If this is true, then I think this is indeed the best way to validate the assumption now. Maybe we can consider to re-evaluate it when we receive the new dataset?

Reply:
Exactly. You are correct, these will be the final timelines when the new dataset is released. But the current timelines is the best estimation I could think of using the current dataset.

Reply:
Great, I think you are totally right indeed!